### PR TITLE
add setting unicast bit

### PR DESCRIPTION
**Pull request by XiXiangFiles**
I have fixed issue for "Support unicast-response bit #21". In the classes.js file, increased 2 lines to support it. I used the string "UNI" to express the "unicast".

### CODE REVIEW

**Comment by pusateri** [comment]
RFC 6762 Section 5.4 calls these "QU" questions for Query Unicast.

**Comment by XiXiangFiles** [comment]
I have renamed the string "UNI" to "QU". thanks for your commit .

**Comment by silverwind** [comment]
Read the RFC section again. I think it certainly is better to handle it like a flag, like `question.QU`, I'd say the property should be absent when decoding a response, given that the RFC states the bit is only valid for questions.

**Comment by XiXiangFiles** [comment]
Many thanks for your comments, i learn a lot in your opinions. i will try to fixed it, and i have an idea for your comments.

**Comment by XiXiangFiles** [comment]
I had revised the codes, the classes.js had recovered to the recent status and the index.js added the flag in question section. When question.QU is "true ", the class value will set the bit.

**Comment by silverwind** [comment]
Could you also add a [test](https://github.com/mafintosh/dns-packet/blob/master/test.js)?

File: index.js [hunk 1]
```diff
@@ -1395,8 +1394,8 @@ question.encode = function (q, buf, offset) {
 
   buf.writeUInt16BE(types.toType(q.type), offset)
   offset += 2
+  if (q.QU === undefined || q.QU !== true) { buf.writeUInt16BE(classes.toClass(q.class === undefined ? 'IN' : q.class), offset) } else { buf.writeUInt16BE(classes.toClass(q.class === undefined ? 'IN' : q.class) + 32768, offset) }
```
Comment (silverwind) [inline index.js]: I think you can reduce this to `if (q.QU)`

Comment (silverwind) [reply index.js]: Also, for the value `32768`, we have the variable `QU_MASK` which you can use.

Comment (silverwind) [reply index.js]: Oh, and please structure the `if` body on a separate line

````js
if (cond) {
 // body
} else {
 // body
}

File: index.js [hunk 2]
```diff
@@ -1417,7 +1416,11 @@ question.decode = function (buf, offset) {
   q.type = types.toString(buf.readUInt16BE(offset))
   offset += 2
 
-  q.class = classes.toString(buf.readUInt16BE(offset))
+  if (buf.readUInt16BE(offset) - 32768 > 0) {
```
Comment (silverwind) [inline index.js]: `buf.readUInt16BE(offset) >= QU_MASK`

File: index.js [hunk 3]
```diff
@@ -1417,7 +1416,11 @@ question.decode = function (buf, offset) {
...
+    classes.toString(buf.readUInt16BE(offset) - 32768)
```
Comment (silverwind) [inline index.js]: I think you want to assign to `q.class` here.

File: index.js (hunk 2 above)
Comment (silverwind) [reply index.js]: also cache the result, e.g. `classValue = buf.readUInt16BE(offset)`.

**Comment by XiXiangFiles** [comment]
I had already added the test and followed your comments to revise the code. thank you very much for your advice.

**Comment by liqinz** [comment]
Is there any progress?

**Comment by ke4nec** [comment]
Is there any progress?

**Comment by MaximKalinin** [comment]
> Should probably raise a new PR with the feedback applied.

What kind of feedback should be applied? I might be wrong, but seems like all the essential suggestions are already there.

**Comment by silverwind** [comment]
These:

https://github.com/mafintosh/dns-packet/pull/49#pullrequestreview-209077128
https://github.com/mafintosh/dns-packet/pull/49#pullrequestreview-209079338
https://github.com/mafintosh/dns-packet/pull/49#pullrequestreview-209079459

Thought it has been far too long, but if you bring the branch up to date, I will have another look.

**Comment by XiXiangFiles** [comment]
@silverwind  update at [here](https://github.com/mafintosh/dns-packet/pull/96)

**Pull request closed without merging**